docs/spec/typeddict.rst

```diff
@@ -128,95 +244,237 @@ key, and the ``'name'`` key is missing::
+matches the extra items type::
+
+    class ExtraMovie(TypedDict, extra_items=bool):
+        name: str
+
+    a: ExtraMovie = {"name": "Blade Runner", "novel_adaptation": True}  # OK
+    b: ExtraMovie = {
+        "name": "Blade Runner",
+        "year": 1982,  # Not OK. 'int' is not assignable to 'bool'
+    }
+
+Here, ``extra_items=bool`` specifies that items other than ``'name'``
+have a value type of ``bool`` and are non-required.
 
+.. _typeddict-inheritance:
 
 Inheritance
-^^^^^^^^^^^
+-----------
 
-It is possible for a TypedDict type to inherit from one or more
-TypedDict types using the class-based syntax.  In this case the
+As discussed under :ref:`typeddict-class-based-syntax`, TypedDict types
+can inherit from one or more other TypedDict types.  In this case the
 ``TypedDict`` base class should not be included.  Example::
 
     class BookBasedMovie(Movie):
         based_on: str
 
 Now ``BookBasedMovie`` has keys ``name``, ``year``, and ``based_on``. It is
-equivalent to this definition, since TypedDict types use :term:`structural`
-:term:`assignability <assignable>`::
+equivalent to this definition, since TypedDict types are :term:`structural` types::
 
     class BookBasedMovie(TypedDict):
         name: str
         year: int
         based_on: str
 
-Here is an example of multiple inheritance::
+Overriding items
+^^^^^^^^^^^^^^^^
 
-    class X(TypedDict):
-        x: int
+Under limited circumstances, subclasses may redeclare items defined in a superclass with
+a different type or different qualifiers. Redeclaring an item with the same type and qualifiers
+is always allowed, although it is redundant.
 
-    class Y(TypedDict):
-        y: str
+If an item is mutable in a superclass, it must remain mutable in the subclass. Similarly,
+mutable items that are :term:`required` in a superclass must remain required in the subclass,
+and mutable :term:`non-required` items in the superclass must remain non-required in the subclass.
+However, if the superclass item is :term:`read-only`, a superclass item that is non-required
+may be overridden with a required item in the subclass. A read-only item in a superclass
+may be redeclared as mutable (that is, without the ``ReadOnly`` qualifier) in a subclass.
+These rules are necessary for type safety.
 
-    class XYZ(X, Y):
-        z: bool
+If an item is read-only in the superclass, the subclass may redeclare it with a different type
+that is :term:`assignable` to the superclass type. Otherwise, changing the type of an item is not allowed.
+Example::
 
-The TypedDict ``XYZ`` has three items: ``x`` (type ``int``), ``y``
-(type ``str``), and ``z`` (type ``bool``).
+   class X(TypedDict):
+       x: str
+       y: ReadOnly[int]
 
-A TypedDict cannot inherit from both a TypedDict type and a
-non-TypedDict base class other than ``Generic``.
+   class Y(X):
+       x: int  # Type check error: cannot overwrite TypedDict field "x"
```

Comment:
I think it would help to (also?) show an example item that is overridden in a covariant way (subclass type assignable to superclass type) but without superclass type being read-only, to clarify that that is still not allowed. The current example might allow the reader to think the override was only disallowed because `int` is not assignable to `str`.

Reply:
Done. I am avoiding the term "covariant" here though; the whole chapter in fact does not mention variance.

Thanks for the review!